ISPN-14028 docs updates to remote jmx connections (#10236)
infinispan/infinispan: documentation/src/main/asciidoc/topics/proc_enabling_jmx_port.adoc

====
endif::remote_caches[]

> You can enable remote JMX ports using one of the following approaches:

Reviewer: I would differentiate this even further:
* Enable remote JMX requiring authentication to one of the Infinispan server security realms
* Enable remote JMX manually, using the standard Java management configuration options

Then, when describing the latter, we can also say that it can be used to set up unsecured JMX remoting, which is not recommended.

====
endif::remote_caches[]

Enabling remote JMX involves exposing sensitive information about your system.

> You can enable remote JMX connections using one of the following approaches:

Reviewer 2: I'm not sure about the first sentence tbh. It sounds like a cautionary note but doesn't quite blend with the following sentence. I would just drop it because we have a warning and other language that advises against the unsecured option.

> * Enable remote JMX connections with authentication.

Reviewer 2: I think it should be "remote JMX ports" not connections. You can enable JMX through the single port `11222` if you want to. Or, if you want a dedicated port for JMX that is more like what you get with WildFly, then you can expose JMX on `9999`.

* Enable remote JMX connections with no authentication or SSL.

.Prerequisites
* For remote JMX with password authentication, you must add the default security realm to your {brandname} configuration.

> Users must have `controlRole` with read/write access or the `monitorRole` with read-only access.

Reviewer: A user with a `controlRole` may use JMX to modify management attribute values (if they are writable) or invoke management operations. Look at https://docs.jboss.org/infinispan/13.0/apidocs/jmxComponents.html#Configuration for a list of these

> Users must have `controlRole` with read/write access or the `monitorRole` with read-only access.

Author: But what is the controlRole? I searched through the repo and I haven't found anything. Do you mean all the roles that have read/write access? https://infinispan.org/docs/stable/titles/configuring/configuring.html#user-roles-permissions_security-authorization

> Users must have `controlRole` with read/write access or the `monitorRole` with read-only access.

Reviewer: No, these are role names the user must belong to. The roles themselves don't need to be defined.
Using the cli:
```
user create --password=admin --groups=admin,monitorRole,controlRole admin
```

.Procedure

> Start the {brandname} Server with remote JMX enabled using one of the following ways:

Reviewer 2: ```suggestion
Start {brandname} Server with a remote JMX port in one of the following ways:
```

** Enable remote JMX connections through port `9999`.
+
[source,options="nowrap",subs=attributes+]
----
bin/server.sh --jmx 9999
----
+
[WARNING]
====
Using remote JMX with SSL disabled is not intended for production environments.
====

** Pass the following system properties to {brandname} Server at startup.
+
[source,options="nowrap",subs=attributes+]
----
bin/server.sh -Dcom.sun.management.jmxremote.port=9999 -Dcom.sun.management.jmxremote.authenticate=false -Dcom.sun.management.jmxremote.ssl=false
----
+
[WARNING]
====

> Enabling remote JMX with no authentication or SSL is insecure and not recommended in any environment.

Reviewer 2: insecure has a different meaning afaik (i.e. lacking in confidence)

> Enabling remote JMX with no authentication or SSL is insecure and not recommended in any environment.

Author: thanks better to say not secure
